sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -55,7 +56,7 @@ object SQLConf {
   val WAREHOUSE_PATH = SQLConfigBuilder("spark.sql.warehouse.dir")
     .doc("The default location for managed databases and tables.")
     .stringConf
-    .createWithDefault("${system:user.dir}/spark-warehouse")
+    .createWithDefault("${system:user.dir}" + File.separator + "spark-warehouse")
```

Comment:
Why not just change the default value to `file:/${system:user.dir}/spark-warehouse`? I do not really understand why we have to change `def warehousePath`.